pkg/transportrequest/uploadcts_test.go

```diff
@@ -0,0 +1,156 @@
+package transportrequest
+
+import (
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestUploadCTS(t *testing.T) {
+
+	fMock := &mock.FilesMock{}
+	files = fMock
+	defer func() { files = piperutils.Files{} }()
+
+	t.Run("npm install command tests", func(t *testing.T) {
+		cmd := mock.ShellMockRunner{}
+		action := CTSUploadAction{
+			Connection:  CTSConnection{Endpoint: "", Client: "", User: "me", Password: "******"},
+			Application: CTSApplication{Pack: "", Name: "", Desc: ""},
+			Node: CTSNode{
+				DeployDependencies: []string{"@sap/my-dep"},
+				InstallOpts:        []string{"--verbose", "--registry", "https://registry.example.org"},
+			},
+			TransportRequestID: "12345678",
+			ConfigFile:         "ui5-deploy.yaml",
+			DeployUser:         "node",
+		}
+		err := action.Perform(&cmd)
+		if assert.NoError(t, err) {
+			assert.Contains(
+				t,
+				cmd.Calls[0],
+				"npm install --global --verbose --registry https://registry.example.org @sap/my-dep",
+			)
+			assert.Contains(
+				t,
+				cmd.Calls[0],
+				"su node",
```

Comment:
Maybe include the new lines `\n` in the check, as the call just containing this is not proof that it is correct, right? Maybe consider checking with a regex multiline pattern such as `^su node$`.

This is true for all cmd.Calls[0] checks in this test file.

Reply:
Or alternatively we could check the entire cmd.Calls[0] entry, to be sure we know what is inside for each test case.

Reply:
Adjusted. Don't prefer to check the full command since that blurs what is important.

Reply:
Here I learned that the testify assert lib also can do regex, that is neat. 👍